### PR TITLE
Add support code for Python 3

### PR DESCRIPTION
**Pull request by roramirez**

### CODE REVIEW

File: starpy/fastagi.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 Module defines a standard Python logging module log 'FastAGI'
 """
+from __future__ import absolute_import
```
Comment (russellb) [inline starpy/fastagi.py]: Is this necessary?  I don't think it is...

File: starpy/manager.py [hunk 2]
```diff
@@ -118,7 +119,10 @@ def registerEvent(self, event, function):
         """
         log.debug('Registering function %s to handle events of type %r',
                   function, event)
-        if isinstance(event, (str, unicode, type(None))):
+        types = [str, type(None)]
+        if sys.version_info <= (3, 0):
+            types.append(unicode)
```
Comment (russellb) [inline starpy/manager.py]: An alternative here would be to use the 'six' library and initialize types using six.string_types.

https://pythonhosted.org/six/

File: starpy/manager.py [hunk 3]
```diff
@@ -134,24 +138,27 @@ def deregisterEvent(self, event, function=None):
         """
         log.debug('Deregistering handler %s for events of type %r',
                   function, event)
-        if isinstance(event, (str, unicode, type(None))):
+        types = [str, type(None)]
+        if sys.version_info <= (3, 0):
+            types.append(unicode)
```
Comment (russellb) [inline starpy/manager.py]: same six.string_types comment here

File: starpy/manager.py [hunk 4]
```diff
@@ -118,7 +119,10 @@ def registerEvent(self, event, function):
         """
         log.debug('Registering function %s to handle events of type %r',
                   function, event)
-        if isinstance(event, (str, unicode, type(None))):
+        types = [str, type(None)]
+        if sys.version_info <= (3, 0):
+            types.append(unicode)
+        if isinstance(event, tuple(types)):
```
Comment (russellb) [inline starpy/manager.py]: I don't think it's necessary to convert to tuple here.

File: starpy/manager.py (hunk 2 above)
Comment (roramirez) [inline starpy/manager.py]: Thats was my first thinking. I wanted to do it without add other dependence to code, this was the end idea.

File: starpy/fastagi.py (hunk 1 above)
Comment (roramirez) [inline starpy/fastagi.py]: Done!

File: starpy/manager.py (hunk 2 above)
Comment (coreyfarrell) [reply starpy/manager.py]: I agree with not adding a new dependency for this one tiny feature.  Maybe I would have created a `def string_types(value):` inside manager.py with different implementation for old vs new to be called from both locations but that is not very important.

In either case I think this PR looks good enough and would prefer to see it merged.

Comment (roramirez) [reply starpy/manager.py]: I'll could would add the one function for this case. Seems good catch to create a string_type method in manager.py. Let me know what your decide.

**Comment by roramirez** [comment]
This PR was merge in https://gerrit.asterisk.org/#/c/8918/

**Pull request closed without merging**